File: 2275-largest-combination-with-bitwise-and-greater-than-zero/2275-largest-combination-with-bitwise-and-greater-than-zero.java
class Solution {
    public int largestCombination(final int[] nums) {
        final int n = nums.length;
        final int m = 24;

        int res = 1;

        for (int bit = 0; bit <= m; bit++) {
            int tmp = 0;
            for (int num : nums) {
                if ((num & (1 << bit)) != 0) {
                    tmp++;
                }
            }
            res = Math.max(res, tmp);
        }
        return res;
    }
}

File: 2275-largest-combination-with-bitwise-and-greater-than-zero/2275-largest-combination-with-bitwise-and-greater-than-zero.rs
impl Solution {
    pub fn largest_combination(nums: Vec<i32>) -> i32 {
        (0..24)
            .map(|bit| nums.iter().filter(|&num| num & (1 << bit) != 0).count() as i32)
            .max()
            .unwrap_or(0)
    }
}
